### PR TITLE
Create GitHub workflows to execute the WinSim demos with libslirp network interface

### PR DESCRIPTION
**Pull request by ChaiTowKway**
Create GitHub workflows to execute the WinSim demos using libslirp network interface
Description
-----------

1. Update Visual Studio project configurations for existing FreeRTOS+TCP demos in P3 that swaps WinPCap with libslirp for user mode networking
2. Create GitHub workflows that execute the FreeRTOS+ demos using these configurations with libslirp on GitHub Runners
3. Connect to Echo server, local MQTT and HTTP server, AWS IOT & S3 service in GitHub workflows for corresponding FreeRTOS+TCP demo execution

-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Skptak — DISMISSED** [review]

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/TCPEchoClient_SingleTasks.c [hunk 1]
```diff
@@ -193,6 +193,12 @@ TickType_t xTimeOnEntering;
 
 				/* Add in some unique text at the front of the string. */
 				sprintf( pcTransmittedString, "TxRx message number %u", ulTxCount );
+
+				/* Replace '\0' with '-' for string length and comparison functions */
+				pcTransmittedString[ strlen( pcTransmittedString ) ] = '-';
+				ulTxCount++;
```
Comment (Skptak) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/TCPEchoClient_SingleTasks.c:199]: Duplicate ulTxCount increase, remove this one or the one on line 202

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/TCPEchoClient_SingleTasks.c [hunk 2]
```diff
@@ -250,12 +256,17 @@ TickType_t xTimeOnEntering;
 					{
 						/* The echo reply was received without error. */
 						ulTxRxCycles[ xInstance ]++;
+
+						/* The "Received correct data" line is used to determine if
+							* this demo runs as part of a GitHub workflow. */
```
Comment (Skptak) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/TCPEchoClient_SingleTasks.c:261]: This comment is misleading - the line is used to determine if the demo runs successfully on a GitHub workflow

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/TCPEchoClient_SingleTasks.c [hunk 3]
```diff
@@ -324,12 +335,15 @@ uint32_t ulRandomNumber;
 		cBuffer[ lCharacter ] = cChar;
 		cChar++;
 
-		if( cChar > '~' )
+		if( cChar > 'Z' )
```
Comment (Skptak) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/TCPEchoClient_SingleTasks.c:338]: Might want to change the cChar to be an "A" if doing this comparison here

File: FreeRTOS-Plus/Demo/corePKCS11_Windows_Simulator/main.c [hunk 4]
```diff
@@ -67,7 +68,7 @@ static void prvPKCS11DemoTask( void * pvParameters )
         vPKCS11SignVerifyDemo();
     #endif
     configPRINTF( ( "---------Finished DEMO---------\r\n" ) );
-
+    Sleep( 3000 );
```
Comment (Skptak) [inline FreeRTOS-Plus/Demo/corePKCS11_Windows_Simulator/main.c:71]: This should be removed

File: FreeRTOS-Plus/Demo/corePKCS11_Windows_Simulator/main.c [hunk 5]
```diff
@@ -30,6 +30,7 @@
 
 /* Standard includes. */
 #include <stdio.h>
+#include <windows.h>
```
Comment (Skptak) [inline FreeRTOS-Plus/Demo/corePKCS11_Windows_Simulator/main.c:33]: This should also be either wrapped in an ifdef for windows, or removed

File: .github/workflows/freertos_plus_demos.yml [hunk 6]
```diff
@@ -112,26 +113,30 @@ jobs:
...
-
       - name: Build HTTP Plain Text Demo
+        id: build-http-plain-text-demo
         working-directory: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_Plaintext
-        run: msbuild http_plain_text_demo.sln -t:rebuild
-
-      - name: Build HTTP S3 Download Demo
-        working-directory: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download
-        run: msbuild http_s3_download_demo.sln -t:rebuild
+        run: msbuild http_plain_text_demo.sln -t:rebuild -property:Configuration=Debug -m
```
Comment (Skptak) [inline .github/workflows/freertos_plus_demos.yml:119]: Looks like this is the wrong solution file?

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/FreeRTOS_Plus_TCP_Minimal.vcxproj [hunk 7]
```diff
@@ -84,8 +84,15 @@
     <Link>
       <SubSystem>Console</SubSystem>
       <GenerateDebugInformation>true</GenerateDebugInformation>
-      <AdditionalDependencies>%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>Iphlpapi.lib;Ws2_32.lib;%(AdditionalDependencies)</AdditionalDependencies>
     </Link>
+    <PostBuildEvent>
+      <Command>xcopy /y /d "..\..\ThirdParty\glib\build\glib\glib-2.0-0.dll" "$(OutDir)"
+xcopy /y /d "..\..\ThirdParty\glib\build\subprojects\proxy-libintl\intl-8.dll" "$(OutDir)"
+
```
Comment (Skptak) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/FreeRTOS_Plus_TCP_Minimal.vcxproj:92]: Are these two lines of whitespace a manually added thing or something from the file?

Comment (ChaiTowKway) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/FreeRTOS_Plus_TCP_Minimal.vcxproj:92]: I think these are something from the file? I added these codes via VS project property.

**Review by Skptak — DISMISSED** [review]

File: FreeRTOS-Plus/ThirdParty/libslirp/.clang-format [hunk 8]
```diff
@@ -0,0 +1,58 @@
+# https://clang.llvm.org/docs/ClangFormat.html
```
Comment (paulbartell) [inline FreeRTOS-Plus/ThirdParty/libslirp/.clang-format:1]: Can libslirp be included as a submodule rather than copied?

Comment (ChaiTowKway) [reply FreeRTOS-Plus/ThirdParty/libslirp/.clang-format]: New commit submitted to submodule libslirp

**Pull request closed without merging**